packages/frontend-v2/hooks/useLocalStorage.ts

```diff
@@ -1,33 +1,38 @@
-import { useState, Dispatch, SetStateAction } from "react";
+import { useState, Dispatch, useEffect } from "react";
 
-// NextJS executes server side code first, and then client side. Window object is only present client side,
-// so window could be undefined. This ensures the window object exists before we use it.
-const isWindow = typeof window !== "undefined";
+// Enum used for formik data, these entries can be deleted later.
+// These are here for testing purposes.
+export enum LocalStorageKey {
+  token = "token"
+}
 
+/**
+ * Using this hook allows for any component to access localstorage.
+ * Accessing the hook allows for the functionality of setting and getting a value in localstorage.
+ * @param key Key for the localstorage value.
+ * @param defaultValue Default value returned if the hook cannot access the value in localstorage.
```

Comment:
pretend that someone reading the comment will only read the comment to figure out what this hook does, and not look at the code (because this is often the case).

~~in this case, "if the hook cannot access the value" is not clear enough for me–you should specify that (1) it~~

sorry i had a better idea midsentence. i think having this hook _not_ take a default value, but then return `null | T` is WAY better design.

- (1) this will FORCE callee to handle the null case, which is good because this is a pretty common use case, and then they can choose exactly how to handle it with a nullcheck, and
- (2) they dont have to understand the weirdness of us only _defaulting_ to the default value, but not actually setting it, etc.

thoughts?